File: src/test/java/com/roscopeco/jasm/model/IntMathTests.java
package com.roscopeco.jasm.model;

public interface IntMathTests {
    byte testI2b(final int in);
    char testI2c(final int in);
    double testI2d(final int in);
    float testI2f(final int in);
    long testI2l(final int in);
    short testI2s(final int in);

    int testIadd(final int d1, final int d2);
    int testIand(final int a, final int b);
    int testIdiv(final int a, final int b);
    int testIinc(final int a);
    int testImul(final int a, final int b);
    int testIneg(final int a);
    int testIor(final int a, final int b);
    int testIrem(final int a, final int b);
    int testIshl(final int a, final int b);
    int testIshr(final int a, final int b);
    int testIsub(final int a, final int b);
    int testIushr(final int a, final int b);
    int testIxor(final int a, final int b);
}

File: src/test/kotlin/com/roscopeco/jasm/model/IntMathTests.kt
package com.roscopeco.jasm.model

interface IntMathTests {
    fun testI2b(input: Int): Byte
    fun testI2c(input: Int): Char
    fun testI2d(input: Int): Double
    fun testI2f(input: Int): Float
    fun testI2l(input: Int): Long
    fun testI2s(input: Int): Short
    fun testIadd(d1: Int, d2: Int): Int
    fun testIand(a: Int, b: Int): Int
    fun testIdiv(a: Int, b: Int): Int
    fun testIinc(a: Int): Int
    fun testImul(a: Int, b: Int): Int
    fun testIneg(a: Int): Int
    fun testIor(a: Int, b: Int): Int
    fun testIrem(a: Int, b: Int): Int
    fun testIshl(a: Int, b: Int): Int
    fun testIshr(a: Int, b: Int): Int
    fun testIsub(a: Int, b: Int): Int
    fun testIushr(a: Int, b: Int): Int
    fun testIxor(a: Int, b: Int): Int
}